File: apple/Sources/R2CCore/ManagedVideoPlaybackStartGate.swift
/// A finite recording starts once per media attempt, never during negotiation
/// or after cancellation. Connection recovery must not rewind that recording.
public struct ManagedVideoPlaybackStartGate: Sendable {
    private var pending = true

    public init() {}

    public mutating func shouldStart(connected: Bool) -> Bool {
        guard pending, connected else { return false }
        pending = false
        return true
    }

    public mutating func cancel() { pending = false }
}

File: app/src/main/java/org/ncssar/rid2caltopo/data/ManagedVideoPlaybackStartGate.java
package org.ncssar.rid2caltopo.data;

/** One finite playback start per media attempt; reconnects do not rewind it. */
public final class ManagedVideoPlaybackStartGate {
    private boolean pending = true;

    public boolean shouldStart(boolean connected) {
        if (!pending || !connected) return false;
        pending = false;
        return true;
    }

    public void cancel() { pending = false; }
}
